packages/grid/x-data-grid-pro/src/components/headerFiltering/GridHeaderFilterAdorment.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import InputAdornment from '@mui/material/InputAdornment';
+import {
+  GridFilterItem,
+  GridFilterOperator,
+  useGridApiContext,
+  GridColDef,
+  useGridRootProps,
+} from '@mui/x-data-grid';
+import { unstable_useId as useId } from '@mui/utils';
+import { unstable_gridHeaderFilteringMenuSelector } from '@mui/x-data-grid/internals';
+import { GridHeaderFilterMenu } from './GridHeaderFilterMenu';
+import { OPERATOR_SYMBOL_MAPPING } from './constants';
+
+function GridHeaderFilterAdorment(props: {
+  operators: GridFilterOperator<any, any, any>[];
+  field: GridColDef['field'];
+  item: GridFilterItem;
+  applyFilterChanges: (item: GridFilterItem) => void;
+  headerFilterMenuRef: React.MutableRefObject<HTMLButtonElement | null>;
+  buttonRef: React.Ref<HTMLButtonElement>;
+}) {
+  const { operators, item, field, buttonRef, headerFilterMenuRef, ...others } = props;
+
+  const buttonId = useId();
+  const menuId = useId();
+
+  const rootProps = useGridRootProps();
+  const apiRef = useGridApiContext();
+  const open = Boolean(
+    unstable_gridHeaderFilteringMenuSelector(apiRef) === field && headerFilterMenuRef.current,
+  );
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    headerFilterMenuRef.current = event.currentTarget;
+    apiRef.current.showHeaderFilterMenu(field);
+  };
+
+  return (
+    <React.Fragment>
+      <InputAdornment position="start">
+        <rootProps.slots.baseIconButton
+          id={buttonId}
+          ref={buttonRef}
+          aria-label={apiRef.current.getLocaleText('filterPanelOperator')}
+          title={apiRef.current.getLocaleText('filterPanelOperator')}
+          aria-controls={menuId}
+          aria-expanded={open ? 'true' : undefined}
+          aria-haspopup="true"
+          tabIndex={-1}
+          size="small"
```

Comment:
It's a broader question but should make the Adorment a slot (or maybe there is already a slot for the entire input including the adornment)?

Reply:
I made this and also menu a slot previously, as you said. Although it should be opinionated, but I decided not to go with it ultimately:
1. It's not just `Adorment` which the user has to reproduce, it's also the logic inside which manages state and encapsulates menu, so probably it's not that practical in real-world scenarios (the same problem that we have with the other filter panel)
2. It's easily achievable using anyways using `colDef.renderHeaderFilter` and also by `slotProps` (it's underway):
```tsx
<DataGridPro
  {...otherProps}
  slotProps={{
    headerFilter: {
      // props passed to root div in `<GridHeaderFilterItem />`,
      ...headerFilterItemPropsOverrides,
      // props passed to respective filter input component
      InputComponentProps: {
        InputProps: {
          startAdorment: <SomeCustomAdorment /> // but it should include menu inside to work like today
        }
      }
    }
  }}
/>
```

What do you think?

Reply:
> Can we move InputAdornment to filter input components, since it's now their implementation detail? (They receive headerFilterMenu prop which is then passed as startAdornment)

Haven't we already moved InputAdornment to the filter input components 🤔  (as `headerFilterMenu` encapsulates both `InputAdornment` and `GridHeaderFilterMenu`) or you mean to extract `InputAdornment` one level above and pass it as a separate prop like `headerFilterMenu`?

Wouldn't it make filter input components less generic and require more Menu related logic to them (as they are also being used a bit differently in filter panel)

Or am I not getting your point correctly 🤔

Reply:
> Also, adding a slot for it would be great

Added slots for Menu, Adornment, and Header filter cell